server/storage/mvcc/kvstore_compaction.go

```diff
@@ -63,6 +64,7 @@ func (s *store) scheduleCompaction(compactMainRev, prevCompactRev int64) (KeyVal
 			// gofail: var compactBeforeSetFinishedCompact struct{}
 			UnsafeSetFinishedCompact(tx, compactMainRev)
 			tx.Unlock()
+			dbCompactionPauseMs.Observe(float64(time.Since(start) / time.Millisecond))
```

Comment:
This seems a separate minor bug fix. The `dbCompactionPauseMs` won't have any data if each time the number of keys to be compacted is less than `batchNum`.

Can we fix this separately ?